pkg/kn/commands/types_test.go

```diff
@@ -141,3 +141,53 @@ func TestGetClientConfig(t *testing.T) {
 		}
 	}
 }
+
+func TestNewSourcesClient(t *testing.T) {
+	basic, err := clientcmd.NewClientConfigFromBytes([]byte(BASIC_KUBECONFIG))
+	namespace := "test"
+	if err != nil {
+		t.Error(err)
+	}
+	for i, tc := range []getConfigTestCase{
+		{
+			clientcmd.NewDefaultClientConfig(clientcmdapi.Config{}, &clientcmd.ConfigOverrides{}),
+			"no configuration has been provided",
+			false,
+		},
+		{
+			basic,
+			"",
+			false,
+		},
+		{ // Test that the cast to wrap the http client in a logger works
+			basic,
+			"",
+			true,
+		},
+	} {
+		p := &KnParams{
```

Comment:
Put this part in a `t.Run()` and include a name of the scenario you are testing?